crates/ty_ide/src/symbols.rs

```diff
@@ -454,12 +764,63 @@ impl SymbolVisitor {
     fn is_constant_name(name: &str) -> bool {
         name.chars().all(|c| c.is_ascii_uppercase() || c == '_')
     }
```

Comment:
Unrelated to this PR but I noticed that you're using it in a few places. It would probalby be good if we have a single definition of what's a constant. E.g. we have something similar (but different) in semantic tokens 

https://github.com/astral-sh/ruff/blob/8f104faa5fb9ef87e06aef822b31f342a177a9d8/crates/ty_ide/src/semantic_tokens.rs#L256-L259

I'm pretty sure we have something similar in Ruff but my search skills are failing me right now.